Migrate remaining audit exceptions to homebrew/core

Rylan12 opened the pull request:
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

This PR migrates the remaining audit exceptions to homebrew/core. Corresponding Homebrew/brew PR: https://github.com/Homebrew/brew/pull/9314

A few notes:

- The `VERSIONED_DEPENDENCIES_CONFLICTS_ALLOWLIST` currently has this message above it: `# TODO: try to remove these, it's not a good user experience`. We can't have comments in JSON files, so this has been removed. Should I open a new GitHub issue for this?
- ~For `VERSIONED_KEG_ONLY_ALLOWLIST`: `cairomm@1.14` was added to Homebrew/brew in https://github.com/Homebrew/brew/pull/9273. However, the homebrew/core PR (https://github.com/Homebrew/homebrew-core/pull/65531) hasn't been merged yet. If it is merge before this, I will re-add `cairomm@1.14`. Until then, though, it cannot be added to the list because that will cause `brew audit` to fail. This will be a one-time problem, though. Part of the reason for moving these lists to homebrew/core is so that issues like this won't occur.~ Done.
- `UNSTABLE_DEVEL_ALLOWLIST` was created to allow for temporary development to occur. Currently, it is empty. We have the option of including an empty JSON file or leaving this file out entirely (although the functionality will remain so it could be added in the future as needed). I opted for adding a blank file because, otherwise, it will probably be very difficult to find next time we need it. If we'd rather leave the file out for now, though, that's fine too.
- These lists are nearly identical to those that existed in Homebrew/brew. The only changes I made were for formulae that have been updated and no longer need to be in the exception list and some minor reordering.
- Lastly, worth noting that this PR will need to be merged into homebrew/core and linuxbrew/core before the brew PR can be merged (due to failing CI), so this will not wait for the brew PR. Once I get approvals in both PRs, this should be merged.

MikeMcQuaid commented:
> * The `VERSIONED_DEPENDENCIES_CONFLICTS_ALLOWLIST` currently has this message above it: `# TODO: try to remove these, it's not a good user experience`. We can't have comments in JSON files, so this has been removed. Should I open a new GitHub issue for this?

Yes, please!

> * I opted for adding a blank file because, otherwise, it will probably be very difficult to find next time we need it.

👍🏻

Rylan12 commented:
Just made a few changes per discussion in https://github.com/Homebrew/brew/pull/9314

1. Added versioned `gcc` formulae to `versioned_keg_only_allowlist.json`
1. Added the TSC as a CODEOWNER for `provided_by_macos_depends_on_allowlist.json`
